### ISSUE TITLE
resourceAwsRoute53RecordCreate issue: terraform-provider-aws: panic: runtime error: invalid memory address or nil pointer dereference

### ISSUE THREAD

**Issue by ckelner** [hashicorp/terraform#3907]
We have a route53 record resource as such:
```
resource "aws_route53_record" "bastion" {
  zone_id = "${var.zone_id}"
  name = "bastion-${var.region}"
  type = "CNAME"
  records = ["${aws_instance.bastion.public_dns}"]
  ttl = "60"
}
```

Elsewhere we have:
```
variable "zone_id" {
  default = ""
}
```

We then set the variable elsewhere for other environments (we are juggling 4-5 applications across 3 AWS accounts, across 5 regions) but the variable `var.zone_id` was missing from configuration for this specific environment.  This produced the crash seen below.  Expected error/output I would expect is for terraform to return some error message from AWS stating that `""` is not a valid zone.

From `crash.log`:
```
2015/11/12 22:01:08 [DEBUG] apply: aws_route53_record.bastion: executing Apply
2015/11/12 22:01:08 [DEBUG] terraform-provider-aws: panic: runtime error: invalid memory address or nil pointer dereference
2015/11/12 22:01:08 [DEBUG] terraform-provider-aws: [signal 0xb code=0x1 addr=0x18 pc=0x527344]
2015/11/12 22:01:08 [DEBUG] terraform-provider-aws: 
2015/11/12 22:01:08 [DEBUG] terraform-provider-aws: goroutine 211 [running]:
2015/11/12 22:01:08 [DEBUG] terraform-provider-aws: github.com/hashicorp/terraform/builtin/providers/aws.resourceAwsRoute53RecordCreate(0xc2081f9d40, 0xbc9360, 0xc208478000, 0x0, 0x0)
2015/11/12 22:01:08 [DEBUG] terraform-provider-aws:     /opt/gopath/src/github.com/hashicorp/terraform/builtin/providers/aws/resource_aws_route53_record.go:137 +0x264
2015/11/12 22:01:08 [DEBUG] terraform-provider-aws: github.com/hashicorp/terraform/helper/schema.(*Resource).Apply(0xc2084dbdc0, 0xc208462090, 0xc2080f9470, 0xbc9360, 0xc208478000, 0x1, 0x0, 0x0)
2015/11/12 22:01:08 [DEBUG] terraform-provider-aws:     /opt/gopath/src/github.com/hashicorp/terraform/helper/schema/resource.go:145 +0x2c1
2015/11/12 22:01:08 [DEBUG] terraform-provider-aws: github.com/hashicorp/terraform/helper/schema.(*Provider).Apply(0xc2083092f0, 0xc208129d40, 0xc208462090, 0xc2080f9470, 0xbca500, 0x0, 0x0)
2015/11/12 22:01:08 [DEBUG] terraform-provider-aws:     /opt/gopath/src/github.com/hashicorp/terraform/helper/schema/provider.go:162 +0x1da
2015/11/12 22:01:08 [DEBUG] terraform-provider-aws: github.com/hashicorp/terraform/rpc.(*ResourceProviderServer).Apply(0xc208377980, 0xc2081ea400, 0xc2080f9520, 0x0, 0x0)
2015/11/12 22:01:08 [DEBUG] terraform-provider-aws:     /opt/gopath/src/github.com/hashicorp/terraform/rpc/resource_provider.go:323 +0x7f
2015/11/12 22:01:08 [DEBUG] terraform-provider-aws: reflect.Value.call(0xdda580, 0x10258e0, 0x13, 0x10aa470, 0x4, 0xc208213f28, 0x3, 0x3, 0x0, 0x0, ...)
2015/11/12 22:01:08 [DEBUG] terraform-provider-aws:     /opt/go/src/reflect/value.go:419 +0x10e5
2015/11/12 22:01:08 [DEBUG] terraform-provider-aws: reflect.Value.Call(0xdda580, 0x10258e0, 0x13, 0xc208213f28, 0x3, 0x3, 0x0, 0x0, 0x0)
2015/11/12 22:01:08 [DEBUG] terraform-provider-aws:     /opt/go/src/reflect/value.go:296 +0xbc
2015/11/12 22:01:08 [DEBUG] terraform-provider-aws: net/rpc.(*service).call(0xc20843c6c0, 0xc20843c680, 0xc208324ee8, 0xc20844a080, 0xc20836c840, 0xbca4a0, 0xc2081ea400, 0x16, 0xbca500, 0xc2080f9520, ...)
2015/11/12 22:01:08 [DEBUG] terraform-provider-aws:     /opt/go/src/net/rpc/server.go:382 +0x1f7
2015/11/12 22:01:08 [DEBUG] terraform-provider-aws: created by net/rpc.(*Server).ServeCodec
2015/11/12 22:01:08 [DEBUG] terraform-provider-aws:     /opt/go/src/net/rpc/server.go:476 +0x44a
2015/11/12 22:01:08 [DEBUG] terraform-provider-aws: 
```

**Comment by ckelner** [comment]
Sorry I should have included the crash.log here (had to change extension to .txt for GitHub to like it.
[crash.log.txt](https://github.com/hashicorp/terraform/files/34217/crash.log.txt)

**Comment by catsby** [comment]
Patched in #3945, thanks for writing in!
